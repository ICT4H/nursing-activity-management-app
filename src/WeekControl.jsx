import React, {Component} from 'react';
import moment from "moment/moment";
import PropTypes from "prop-types";
import WeeklyTable from "./WeeklyTable";

class WeekControl extends Component {
  render() {
    return (
        <div className={this.props.className}>
          <button onClick={this.props.pastWeek} className={"pastWeek"}>{"<"}</button>
          <p>{moment(this.props.currentWeek.startingDate).format('DD')}-
            {moment(this.props.currentWeek.endingDate).format('DD MMM')}
          </p>
          <button onClick={this.props.nextWeek} className={"nextWeek"}>{">"}</button>
        </div>
    )
  }
}

WeeklyTable.propTypes = {
  currentWeek: PropTypes.objectOf(PropTypes.instanceOf(Date)).isRequired,
  date: PropTypes.arrayOf(PropTypes.object)
};

export default WeekControl
