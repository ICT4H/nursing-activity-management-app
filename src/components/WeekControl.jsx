import React, {Component} from 'react';
import moment from "moment/moment";
import PropTypes from "prop-types";

class WeekControl extends Component {
  render() {
    const currentWeek = this.props.currentWeek;
    return (
        <div className={this.props.className}>
          <button onClick={this.props.pastWeek} className={"pastWeek"}>{"<"}</button>
          <div>{moment(currentWeek.startingDate).format('DD')}-
            {moment(currentWeek.endingDate).format('DD MMM')}
          </div>
          <button onClick={this.props.nextWeek} className={"nextWeek"}>{">"}</button>
        </div>
    )
  }
}

WeekControl.propTypes = {
  currentWeek: PropTypes.objectOf(PropTypes.instanceOf(Date)).isRequired,
  date: PropTypes.arrayOf(PropTypes.object)
};

export default WeekControl
