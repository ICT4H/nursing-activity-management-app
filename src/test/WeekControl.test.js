import React from 'react';
import WeekControl from '../WeekControl';
import renderer from 'react-test-renderer';
import moment from "moment/moment";

test('Should have current week starting date and ending date', () => {
  const currentWeek = {
    startingDate: moment(new Date('June 5, 2018 9:30:00')).day(0).toDate(),
    endingDate: moment(new Date('June 5, 2018 9:30:00')).day(6).toDate()
  };
  let component=renderer.create(<WeekControl currentWeek={currentWeek}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should function pastWeek called when clicked on pastWeek button', () => {
  const currentWeek = {
    startingDate: moment(new Date('June 5, 2018 9:30:00')).day(0).toDate(),
    endingDate: moment(new Date('June 5, 2018 9:30:00')).day(6).toDate()
  };
  const mockPastWeekFn = jest.fn();

  let component=renderer.create(<WeekControl currentWeek={currentWeek} pastWeek={mockPastWeekFn}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let pastWeekButton = (tree.children.find((c)=>c.className="pastWeek"));
  expect(mockPastWeekFn.mock.calls.length).toBe(0);
  pastWeekButton.props.onClick();
  expect(mockPastWeekFn.mock.calls.length).toBe(1);

  pastWeekButton.props.onClick();
  expect(mockPastWeekFn.mock.calls.length).toBe(2);
});