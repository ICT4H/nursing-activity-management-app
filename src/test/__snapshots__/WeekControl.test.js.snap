// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should function pastWeek called when clicked on pastWeek button 1`] = `
<div
  className={undefined}
>
  <button
    className="pastWeek"
    onClick={[Function]}
  >
    &lt;
  </button>
  <p>
    03
    -
    09 Jun
  </p>
  <button
    className="nextWeek"
    onClick={undefined}
  >
    &gt;
  </button>
</div>
`;

exports[`Should have current week starting date and ending date 1`] = `
<div
  className={undefined}
>
  <button
    className="pastWeek"
    onClick={undefined}
  >
    &lt;
  </button>
  <p>
    03
    -
    09 Jun
  </p>
  <button
    className="nextWeek"
    onClick={undefined}
  >
    &gt;
  </button>
</div>
`;
